adapters/handlers/rest/clusterapi/indices_replicas.go

```diff
@@ -883,3 +982,47 @@ func localIndexNotReady(resp replica.SimpleResponse) bool {
 	}
 	return false
 }
+
+// Close gracefully shuts down the replicatedIndices by draining the queue and waiting for workers to finish
+func (i *replicatedIndices) Close(ctx context.Context) error {
```

Comment:
let's just add a debug log message here that the shutdown started+completed. Just in case it hangs for whatever reason :) Then we can merge